sdk/storage/azure-storage-internal-avro/src/main/java/com/azure/storage/internal/avro/implementation/AvroParser.java

```diff
@@ -0,0 +1,112 @@
+package com.azure.storage.internal.avro.implementation;
+
+import com.azure.storage.internal.avro.implementation.schema.AvroSchema;
+import com.azure.storage.internal.avro.implementation.schema.AvroType;
+import com.azure.storage.internal.avro.implementation.schema.file.AvroBlockSchema;
+import com.azure.storage.internal.avro.implementation.schema.file.AvroHeaderSchema;
+import reactor.core.publisher.Flux;
+
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * A class that represents a push based AvroParser that can parse avro data from a reactive stream.
+ *
+ * The parser stores the {@link AvroParserState current state}, the sync marker (parsed from the header),
```

Comment:
Can you elaborate on what the sync marker and file type are either here or on their field declaration?